source/server/configuration_impl.cc

```diff
@@ -38,60 +38,15 @@ bool FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager
   return filter_manager.initializeReadFilters();
 }
 
-/**
- * All missing listener config stats. @see stats_macros.h
- */
-#define ALL_MISSING_LISTENER_CONFIG_STATS(COUNTER) COUNTER(extension_config_missing)
-/**
- * Struct definition for all missing listener config stats. @see stats_macros.h
- */
-struct MissingListenerConfigStats {
-  ALL_MISSING_LISTENER_CONFIG_STATS(GENERATE_COUNTER_STRUCT)
-};
-
-class MissingConfigTcpListenerFilter : public Network::ListenerFilter,
-                                       public Logger::Loggable<Logger::Id::filter> {
-public:
-  MissingConfigTcpListenerFilter(Stats::Scope& stats_scope)
-      : scope_(stats_scope), stats_({ALL_MISSING_LISTENER_CONFIG_STATS(POOL_COUNTER(scope_))}) {}
-
-  // Network::ListenerFilter
-  Network::FilterStatus onAccept(Network::ListenerFilterCallbacks& cb) override {
-    ENVOY_LOG(debug, "Listener filter: new connection accepted while missing configuration. "
-                     "Close socket and stop the iteration onAccept.");
-    cb.socket().ioHandle().close();
-    stats_.extension_config_missing_.inc();
-    return Network::FilterStatus::StopIteration;
-  }
-  Network::FilterStatus onData(Network::ListenerFilterBuffer&) override {
-    // The socket is already closed onAccept. Just return StopIteration here.
-    return Network::FilterStatus::StopIteration;
-  }
-  size_t maxReadBytes() const override { return 0; }
-
-private:
-  Stats::Scope& scope_;
-  MissingListenerConfigStats stats_;
-};
-
 bool FilterChainUtility::buildFilterChain(Network::ListenerFilterManager& filter_manager,
-                                          const Filter::ListenerFilterFactoriesList& factories,
-                                          Stats::Scope& stats_scope) {
-  bool added_missing_config_filter = false;
+                                          const Filter::ListenerFilterFactoriesList& factories) {
   for (const auto& filter_config_provider : factories) {
     auto config = filter_config_provider->config();
-    if (config.has_value()) {
-      auto config_value = config.value();
-      config_value(filter_manager);
-      continue;
-    }
-
-    // If a filter config is missing after warming, stop iteration.
-    if (!added_missing_config_filter) {
-      filter_manager.addAcceptFilter(nullptr,
-                                     std::make_unique<MissingConfigTcpListenerFilter>(stats_scope));
-      added_missing_config_filter = true;
+    if (!config.has_value()) {
```

Comment:
Thanks for taking care of this!

We are working on the UDP listener ECDS now. IIUC, we need to keep this logic consistent there, i.e, directly return in this function if the dynamic config is missing: https://github.com/envoyproxy/envoy/blob/97d599f0bda7ca2a4402f7ef0902484a602492ab/source/server/configuration_impl.cc#L100

Currently buildUdpFilterChain() return void.  Should we change it into return bool, and install same logic here to keep them consistent?
https://github.com/envoyproxy/envoy/blob/97d599f0bda7ca2a4402f7ef0902484a602492ab/source/server/active_udp_listener.cc#L102

Reply:
Checking the code, the UDP is different from TCP. The TCP will create new filter factories for the new connection, the UDP is different.

https://github.com/envoyproxy/envoy/blob/75aa3fbac7a72d27e6a7a003ddfeac15883e36e2/source/server/active_udp_listener.cc#L102

I'm curious how do you implement ecds for udp, since the filter factories is created in the beginning of listener created, if the filter factories begin with a invalid config, then you have no chance update the filter factories.

Reply:
Saw @yanjunxiang-google submitted a draft for UDP ecds https://github.com/envoyproxy/envoy/pull/21606

If we enable to trigger the udp listener filter factory to recreate when config updated, then we should be able to do the same thing as the TCP doing, just reject the connection, when an empty config was updated.